### DISCUSSION TITLE
FlxSprites with a negative scale stop rendering when reaching the edge of a camera

### DISCUSSION THREAD

**Discussion by mayo78** [HaxeFlixel/flixel#2836] [Q&A] [answered]
- **Haxe version:** 4.2.5
- **Flixel version:** 5.3.1
- **OpenFL version:** 9.2.1
- **Lime version:** 8.0.1
- **Affected targets:** probably all?

________________________________

**Code snippet reproducing the issue**:

```haxe
package;

import flixel.util.FlxColor;
import flixel.FlxG;
import flixel.FlxSprite;
import flixel.FlxState;

//press a to swap from an x scale of 1 and -1, red is 1 blue is -1

class PlayState extends FlxState
{
	var hi:FlxSprite;
	override public function create()
	{
		hi = new FlxSprite().makeGraphic(100, 100, FlxColor.WHITE);
		hi.color = FlxColor.RED;
		add(hi);
		super.create();
	}

	override public function update(elapsed:Float)
	{
		hi.setPosition(FlxG.mouse.x, FlxG.mouse.y);
		if(FlxG.keys.justPressed.A) {
			hi.scale.x *= -1;
			hi.color = (hi.scale.x == -1) ? FlxColor.BLUE : FlxColor.RED;
			hi.updateHitbox();	
		}
		super.update(elapsed);
	}
}

```

________________________________

**Observed behavior:** the sprite stops rendering at the edge of the camera when it should be visible

**Expected behavior:** the sprite stops rendering when it leaves the cameras area

https://github.com/HaxeFlixel/flixel/assets/75394229/7dc5cfa8-54e5-41a6-88b2-c0830c18f1c7

**Comment by Geokureli** [comment, marked as answer]
Scale should not be negative, if you want to invert a sprite, use the flipX and flipY fields

Ill see if this can be handled better, but negative scale may cause other issues elsewhere, and should really be avoided
